Sources/SwiftGenKit/Parsers/AssetsCatalogParser.swift

```diff
@@ -80,7 +80,7 @@ extension AssetsCatalogParser {
    This method recursively parses a directory structure, processing each folder (files are ignored).
   */
   func process(folder: Path, withPrefix prefix: String = "") -> [Catalog.Entry] {
-    return (try? folder.children().sorted(by: <).flatMap {
+    return (try? folder.children().sorted(by: <).compactMap {
```

Comment:
Not sure I get this change

`folder.children().sorted(by:<)` is an array
the closure calls `process` which returns an array too

So in this use case we should be using the monadic flatMap (the one that wasn't renamed), not the non-monadic `[T].compactMap(f: (T) -> T?) -> [T]` one? So why would the compiler want us to change it there, did we get types mixed up and didn't do what we thought we were doing before, using the non monadic one all along by mistake?

Reply:
The `process(item: $0, withPrefix: prefix)` that's called in the `compactMap` closure has as type:
```swift
func process(item: Path, withPrefix prefix: String) -> Catalog.Entry?
```

Reply:
Oh right it's not the same `process` method as `process(folder: withPrefix: ) -> [Catalog.Entry]`, right. 👍